[visionOS] Re-enable CALayer/UIView Tree based tests and make them more reliable

etiennesegonzac opened the pull request:
#### 0c009bd3041dcffddadd9656a928076ba262462f
<pre>
[visionOS] Re-enable CALayer/UIView Tree based tests and make them more reliable
<a href="https://bugs.webkit.org/show_bug.cgi?id=303110">https://bugs.webkit.org/show_bug.cgi?id=303110</a>
&lt;<a href="https://rdar.apple.com/165410344">rdar://165410344</a>&gt;

Reviewed by Mike Wyrzykowski.

Introduce new test-only helper to get the text representation of a
CALayer or UIView subtree. This makes the tests much more focused and
easier to maintain, since they don&apos;t pick-up unrelated changes.

The methods are called `UIHelper.getUIViewTreeForCompositedElement()`
and `UIHelper.getCALayerTreeForCompositedElement()` to make it clear
that the element needs its own layer for this to work.

* LayoutTests/interaction-region/guard-crash-expected.txt:
* LayoutTests/interaction-region/guard-crash.html:
* LayoutTests/interaction-region/interaction-layers-culling-expected.txt:
* LayoutTests/interaction-region/interaction-layers-culling-layer-type-change-expected.txt:
* LayoutTests/interaction-region/interaction-layers-culling-layer-type-change.html:
* LayoutTests/interaction-region/interaction-layers-culling.html:
* LayoutTests/interaction-region/layer-tree-expected.txt:
* LayoutTests/interaction-region/layer-tree-shape-reset-expected.txt:
* LayoutTests/interaction-region/layer-tree-shape-reset.html:
* LayoutTests/interaction-region/layer-tree.html:
Update the interaction-region CALayer tree based test to use the new
method, focused on the #test element.

* LayoutTests/platform/visionos/TestExpectations:
Re-enabled the tests.

* LayoutTests/platform/visionos/transforms/separated-expected.txt:
* LayoutTests/platform/visionos/transforms/separated-image-animated-expected.txt:
* LayoutTests/platform/visionos/transforms/separated-image-animated.html:
* LayoutTests/platform/visionos/transforms/separated-image-downgrade-content-expected.txt:
* LayoutTests/platform/visionos/transforms/separated-image-downgrade-content.html:
* LayoutTests/platform/visionos/transforms/separated-image-downgrade-expected.txt:
* LayoutTests/platform/visionos/transforms/separated-image-downgrade-tiled-expected.txt:
* LayoutTests/platform/visionos/transforms/separated-image-downgrade-tiled.html:
* LayoutTests/platform/visionos/transforms/separated-image-downgrade.html:
* LayoutTests/platform/visionos/transforms/separated-image-upgrade-expected.txt:
* LayoutTests/platform/visionos/transforms/separated-image-upgrade-tiled-expected.txt:
* LayoutTests/platform/visionos/transforms/separated-image-upgrade-tiled.html:
* LayoutTests/platform/visionos/transforms/separated-image-upgrade.html:
* LayoutTests/platform/visionos/transforms/separated-update-expected.txt:
* LayoutTests/platform/visionos/transforms/separated-update.html:
* LayoutTests/platform/visionos/transforms/separated-video-expected.txt:
* LayoutTests/platform/visionos/transforms/separated-video.html:
* LayoutTests/platform/visionos/transforms/separated.html:
Update the transforms UIView/CALayer tree based test to use the new
method, focused on the #test element.

* LayoutTests/resources/ui-helper.js:
(window.UIHelper.getUIViewTreeForCompositedElement):
(window.UIHelper.getCALayerTreeForCompositedElement):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivateForTesting.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewTesting.mm:
(-[WKWebView _caLayerTreeAsTextForLayerWithID:]):
(-[WKWebView _caLayerTreeAsTextForLayer:]):
(-[WKWebView _caLayerTreeAsText]):
* Source/WebKit/UIProcess/API/ios/WKWebViewPrivateForTestingIOS.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewTestingIOS.mm:
(dumpUIView):
(-[WKWebView _uiViewTreeAsText]):
(-[WKWebView _uiViewTreeAsTextForViewWithLayerID:]):
(-[WKWebView _uiViewTreeAsTextForView:]):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
Make the `m_remoteLayerTreeHost` protected instead of private for the
test-only method in `RemoteLayerTreeDrawingAreaProxyIOS`.
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeDrawingAreaProxyIOS.h:
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeDrawingAreaProxyIOS.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxyIOS::viewWithLayerIDForTesting const):
* Tools/TestRunnerShared/UIScriptContext/Bindings/UIScriptController.idl:
* Tools/TestRunnerShared/UIScriptContext/UIScriptController.h:
(WTR::UIScriptController::caLayerTreeAsTextForLayerWithID const):
(WTR::UIScriptController::uiViewTreeAsTextForViewWithLayerID const):
* Tools/WebKitTestRunner/cocoa/UIScriptControllerCocoa.h:
* Tools/WebKitTestRunner/cocoa/UIScriptControllerCocoa.mm:
(WTR::UIScriptControllerCocoa::caLayerTreeAsTextForLayerWithID const):
* Tools/WebKitTestRunner/ios/UIScriptControllerIOS.h:
* Tools/WebKitTestRunner/ios/UIScriptControllerIOS.mm:
(WTR::UIScriptControllerIOS::uiViewTreeAsTextForViewWithLayerID const):
Add new test-only methods to get the text representation of a CALayer or
UIView subtree based on a Platform Layer ID.

Canonical link: <a href="https://commits.webkit.org/303600@main">https://commits.webkit.org/303600@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/19fb830effa29185e3fa7e1dcf48f8bee0099086

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/132948 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/5449 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/44040 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/140483 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/84979 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/134818 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/5867 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/5312 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/101662 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/68992 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/05742e17-72d3-417d-935a-5c7c72aa9bd5) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/135894 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/4118 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/119122 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82462 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/4c7a4076-6f9b-4b91-98c4-84ba60ab36fe) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/4013 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/1626 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/83716 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/113047 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/37242 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/143136 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/5118 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/37821 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/110039 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/5200 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/4384 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/110220 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27945 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3928 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/115384 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/58688 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/5172 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/33739 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/5012 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/68624 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/5262 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/5130 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski reviewed (approved):
iOS crashes appear unrelated:
```

Exception Type:    EXC_CRASH (SIGKILL)
Exception Codes:   0x0000000000000000, 0x0000000000000000

Termination Reason:  Namespace LIBXPC, Code 4, XPC_EXIT_REASON_SIGTERM_TIMEOUT


Thread 0 Crashed:
0   libsystem_kernel.dylib        	       0x100974b70 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x10098590c mach_msg2_internal + 72
2   libsystem_kernel.dylib        	       0x10097cc10 mach_msg_overwrite + 480
3   libsystem_kernel.dylib        	       0x100974ee4 mach_msg + 20
4   CoreFoundation                	       0x18044cd3c __CFRunLoopServiceMachPort + 156
5   CoreFoundation                	       0x18044bef4 __CFRunLoopRun + 1128
6   CoreFoundation                	       0x180446e24 _CFRunLoopRunSpecificWithOptions + 496
7   Foundation                    	       0x1810f1cac -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 208
8   Foundation                    	       0x1810f1e7c -[NSRunLoop(NSRunLoop) run] + 60
9   akd                           	       0x1004889c0 0x10041c000 + 444864
```

etiennesegonzac committed:
[visionOS] Re-enable CALayer/UIView Tree based tests and make them more reliable
https://bugs.webkit.org/show_bug.cgi?id=303110
<rdar://165410344>

Reviewed by Mike Wyrzykowski.

Introduce new test-only helper to get the text representation of a
CALayer or UIView subtree. This makes the tests much more focused and
easier to maintain, since they don't pick-up unrelated changes.

The methods are called `UIHelper.getUIViewTreeForCompositedElement()`
and `UIHelper.getCALayerTreeForCompositedElement()` to make it clear
that the element needs its own layer for this to work.

* LayoutTests/interaction-region/guard-crash-expected.txt:
* LayoutTests/interaction-region/guard-crash.html:
* LayoutTests/interaction-region/interaction-layers-culling-expected.txt:
* LayoutTests/interaction-region/interaction-layers-culling-layer-type-change-expected.txt:
* LayoutTests/interaction-region/interaction-layers-culling-layer-type-change.html:
* LayoutTests/interaction-region/interaction-layers-culling.html:
* LayoutTests/interaction-region/layer-tree-expected.txt:
* LayoutTests/interaction-region/layer-tree-shape-reset-expected.txt:
* LayoutTests/interaction-region/layer-tree-shape-reset.html:
* LayoutTests/interaction-region/layer-tree.html:
Update the interaction-region CALayer tree based test to use the new
method, focused on the #test element.

* LayoutTests/platform/visionos/TestExpectations:
Re-enabled the tests.

* LayoutTests/platform/visionos/transforms/separated-expected.txt:
* LayoutTests/platform/visionos/transforms/separated-image-animated-expected.txt:
* LayoutTests/platform/visionos/transforms/separated-image-animated.html:
* LayoutTests/platform/visionos/transforms/separated-image-downgrade-content-expected.txt:
* LayoutTests/platform/visionos/transforms/separated-image-downgrade-content.html:
* LayoutTests/platform/visionos/transforms/separated-image-downgrade-expected.txt:
* LayoutTests/platform/visionos/transforms/separated-image-downgrade-tiled-expected.txt:
* LayoutTests/platform/visionos/transforms/separated-image-downgrade-tiled.html:
* LayoutTests/platform/visionos/transforms/separated-image-downgrade.html:
* LayoutTests/platform/visionos/transforms/separated-image-upgrade-expected.txt:
* LayoutTests/platform/visionos/transforms/separated-image-upgrade-tiled-expected.txt:
* LayoutTests/platform/visionos/transforms/separated-image-upgrade-tiled.html:
* LayoutTests/platform/visionos/transforms/separated-image-upgrade.html:
* LayoutTests/platform/visionos/transforms/separated-update-expected.txt:
* LayoutTests/platform/visionos/transforms/separated-update.html:
* LayoutTests/platform/visionos/transforms/separated-video-expected.txt:
* LayoutTests/platform/visionos/transforms/separated-video.html:
* LayoutTests/platform/visionos/transforms/separated.html:
Update the transforms UIView/CALayer tree based test to use the new
method, focused on the #test element.

* LayoutTests/resources/ui-helper.js:
(window.UIHelper.getUIViewTreeForCompositedElement):
(window.UIHelper.getCALayerTreeForCompositedElement):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivateForTesting.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewTesting.mm:
(-[WKWebView _caLayerTreeAsTextForLayerWithID:]):
(-[WKWebView _caLayerTreeAsTextForLayer:]):
(-[WKWebView _caLayerTreeAsText]):
* Source/WebKit/UIProcess/API/ios/WKWebViewPrivateForTestingIOS.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewTestingIOS.mm:
(dumpUIView):
(-[WKWebView _uiViewTreeAsText]):
(-[WKWebView _uiViewTreeAsTextForViewWithLayerID:]):
(-[WKWebView _uiViewTreeAsTextForView:]):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
Make the `m_remoteLayerTreeHost` protected instead of private for the
test-only method in `RemoteLayerTreeDrawingAreaProxyIOS`.
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeDrawingAreaProxyIOS.h:
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeDrawingAreaProxyIOS.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxyIOS::viewWithLayerIDForTesting const):
* Tools/TestRunnerShared/UIScriptContext/Bindings/UIScriptController.idl:
* Tools/TestRunnerShared/UIScriptContext/UIScriptController.h:
(WTR::UIScriptController::caLayerTreeAsTextForLayerWithID const):
(WTR::UIScriptController::uiViewTreeAsTextForViewWithLayerID const):
* Tools/WebKitTestRunner/cocoa/UIScriptControllerCocoa.h:
* Tools/WebKitTestRunner/cocoa/UIScriptControllerCocoa.mm:
(WTR::UIScriptControllerCocoa::caLayerTreeAsTextForLayerWithID const):
* Tools/WebKitTestRunner/ios/UIScriptControllerIOS.h:
* Tools/WebKitTestRunner/ios/UIScriptControllerIOS.mm:
(WTR::UIScriptControllerIOS::uiViewTreeAsTextForViewWithLayerID const):
Add new test-only methods to get the text representation of a CALayer or
UIView subtree based on a Platform Layer ID.

Canonical link: https://commits.webkit.org/303600@main
